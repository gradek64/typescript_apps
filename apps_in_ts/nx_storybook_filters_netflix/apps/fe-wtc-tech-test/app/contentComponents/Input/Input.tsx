import React, { useState } from 'react';

interface PropsI {
  value: string | number;
  checked?: boolean;
  name: string;
  callback: (state: {}) => void;
}
function Input({ value, checked = false, callback, name }: PropsI) {
  const [checkedState, setcheckedState] = useState(checked);
  return (
    <label className="container-checkbox">
      <input
        name={name}
        type="checkbox"
        checked={checkedState}
        onChange={() => {
          setcheckedState(!checkedState);
          callback({ [name]: !checkedState });
        }}
      />
      {value}
      <span className="checkmark"></span>
    </label>
  );
}

export default Input;
