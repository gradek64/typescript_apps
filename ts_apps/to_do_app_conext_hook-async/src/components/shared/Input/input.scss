label.container-checkbox {
  min-width: 65px;
}
