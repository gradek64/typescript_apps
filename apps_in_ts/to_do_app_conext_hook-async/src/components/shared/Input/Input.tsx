import React, { useState } from 'react';

//custom
import './input.scss';

interface PropsI {
  value: string | number;
  checked?: boolean;
  name: string;
  disable?: boolean;
  callback: (state: {}) => void;
}
function Input({
  value,
  checked = false,
  disable = false,
  callback,
  name,
}: PropsI) {
  const [checkedState, setcheckedState] = useState(checked);
  return (
    <label className="container-checkbox">
      <input
        name={name}
        type="checkbox"
        disabled={disable}
        checked={checkedState}
        onChange={() => {
          setcheckedState(!checkedState);
          callback({ id: name, status: !checkedState });
        }}
      />
      {value}
      <span className="checkmark"></span>
    </label>
  );
}

export default Input;
