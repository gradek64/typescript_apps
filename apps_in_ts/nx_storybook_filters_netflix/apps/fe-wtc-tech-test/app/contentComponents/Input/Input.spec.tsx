import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from './Input';
import { Simulate } from 'react-dom/test-utils';

const props = {
  callback: jest.fn(),
  name: 'watch',
  value: 4,
};

describe('should render correclty with initial props', () => {
  afterEach(cleanup);
  it('should render and take a snapshot', () => {
    const { asFragment } = render(<Input {...props} />);
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });
});
describe('callback CTA', () => {
  afterEach(cleanup);
  it('should call callback on onChange event', () => {
    const { container } = render(<Input {...props} />);
    const input = container.querySelector('input');
    /* confirmed to have bug
    fireEvent.change(input, {
      target: { value: { watch: true } },
    }); */
    Simulate.change(input);
    expect(input.checked).toEqual(true);
    expect(props.callback).toBeCalledWith({ watch: true });

    Simulate.change(input);
    expect(input.checked).toEqual(false);
    expect(props.callback).toBeCalledWith({ watch: false });
  });
});
