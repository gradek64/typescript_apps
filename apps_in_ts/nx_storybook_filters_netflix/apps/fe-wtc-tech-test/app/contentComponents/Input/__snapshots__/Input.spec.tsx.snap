// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty with initial props should render and take a snapshot 1`] = `
<DocumentFragment>
  <label
    class="container-checkbox"
  >
    <input
      name="watch"
      type="checkbox"
    />
    4
    <span
      class="checkmark"
    />
  </label>
</DocumentFragment>
`;
